src/transaction.rs

```diff
@@ -182,31 +183,43 @@ impl Transaction {
             .unique()
     }
 
-    pub const fn gas_price(&self) -> Word {
+    pub const fn gas_price(&self) -> Option<Word> {
         match self {
-            Self::Script { gas_price, .. } => *gas_price,
-            Self::Create { gas_price, .. } => *gas_price,
+            Self::Script { gas_price, .. } => Some(*gas_price),
+            Self::Create { .. } => None,
         }
     }
 
     pub fn set_gas_price(&mut self, price: Word) {
-        match self {
-            Self::Script { gas_price, .. } => *gas_price = price,
-            Self::Create { gas_price, .. } => *gas_price = price,
+        if let Self::Script { gas_price, .. } = self {
+            *gas_price = price
         }
     }
 
-    pub const fn gas_limit(&self) -> Word {
+    pub const fn gas_limit(&self) -> Option<Word> {
         match self {
-            Self::Script { gas_limit, .. } => *gas_limit,
-            Self::Create { gas_limit, .. } => *gas_limit,
+            Self::Script { gas_limit, .. } => Some(*gas_limit),
+            Self::Create { .. } => None,
         }
     }
 
     pub fn set_gas_limit(&mut self, limit: Word) {
+        if let Self::Script { gas_limit, .. } = self {
+            *gas_limit = limit
+        }
```

Comment:
Hmmm. Should this just pass silently if you're trying to set the gas price of a contract creation tx? Technically that logic should be unreachable _assuming the rest of the code is bug-free_, but should probably have some sort of error here since the rest of the code may not be bug-free.